api/src/movies/dao/movies.dao.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable } from '@nestjs/common';
+import { v4 as uuidv4 } from 'uuid';
+import { PrismaService } from '../../prisma/prisma.service';
+import { CreateListDto } from '../../lists/dtos/create-list.dto';
+
+@Injectable()
+export class MoviesDao {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async createMovies(movies: CreateListDto, listId: string) {
```

Comment:
This should take an array of movies, not a `CreateListDto` type object. The `Movie` module (including service, dao, types, guards, etc) shouldn't know anything about lists. The only external elements that will be used in the `Movie` are ids, like `listId` and `userId`, which are both strings, so no `Movie` or `User` type should be present in this module at all.

Reply:
Agreed, should be fixed.